### PR TITLE
DMP-233

### PR DESCRIPTION
**Pull request by HMCTSDeveloper**
Changes based of soap gateway

### CODE REVIEW

**Review by cmagowan-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java [hunk 1]
```diff
@@ -54,30 +53,42 @@ public class DailyListProcessorImpl implements DailyListProcessor {
...
+                    try {
+                        processDailyList(dailyLists.get(0));
+                    } catch (JsonProcessingException e) {
+                        dailyLists.get(0).setStatus(String.valueOf(JobStatusType.FAILED));
+                        log.error("Failed to process dailylist for courthouse: {} with dailylist id: {}",
+                                  courthouse, dailyLists.get(0).getId(), e
+                        );
+                    }
+                }
+                if (dailyLists.size() > 1) {
```
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java:87]: should this be inside the `if (!dailyLists.isEmpty()) {` loop, as if `dailyLists` is empty then don't need to evaluate the size?

File: src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java [hunk 2]
```diff
@@ -54,30 +53,42 @@ public class DailyListProcessorImpl implements DailyListProcessor {
...
+                    date, String.valueOf(source)
+                );
+                // Daily lists are being ordered descending by date so first item will be the most recent version
+                if (!dailyLists.isEmpty()) {
+                    try {
+                        processDailyList(dailyLists.get(0));
+                    } catch (JsonProcessingException e) {
+                        dailyLists.get(0).setStatus(String.valueOf(JobStatusType.FAILED));
+                        log.error("Failed to process dailylist for courthouse: {} with dailylist id: {}",
+                                  courthouse, dailyLists.get(0).getId(), e
```
Comment (cmagowan-hmcts) [inline src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java:83]: this is the courthouse object, we might want to use its name or something instead?

Comment (HMCTSDeveloper) [reply src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java:83]: ACCEPT.

File: src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java (hunk 1 above)
Comment (HMCTSDeveloper) [reply src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java:87]: ACCEPT.

**Review by cmagowan-hmcts — APPROVED** [review]

**Pull request merged**